### PR TITLE
Fix 0-length alloc in dynamic table

### PR DESCRIPTION
**Pull request by ColdenCullen**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ColdenCullen** [comment]
See awslabs/aws-c-common#469

File: source/hpack.c [hunk 1]
```diff
@@ -277,9 +277,11 @@ struct aws_hpack_context *aws_hpack_context_new(struct aws_allocator *allocator,
     aws_huffman_decoder_init(&context->decoder, hpack_coder);
 
     /* Initialize dynamic table */
-    const size_t buffer_size = max_dynamic_elements * sizeof(struct aws_http_header);
-    context->dynamic_table.buffer = aws_mem_acquire(allocator, buffer_size);
-    memset(context->dynamic_table.buffer, 0, buffer_size);
+    if (max_dynamic_elements) {
+        const size_t buffer_size = max_dynamic_elements * sizeof(struct aws_http_header);
+        context->dynamic_table.buffer = aws_mem_acquire(allocator, buffer_size);
```
Comment (JonathanHenson) [inline source/hpack.c]: aws_mem_calloc and lose the memset?

Comment (ColdenCullen) [reply source/hpack.c]: S M O R T

Comment (justinboswell) [reply source/hpack.c]: +1

**Comment by ColdenCullen** [comment]
> Do you have tests to exercise these code paths?

Apparently, see the builds that failed before I pushed the most recent commit :P

File: source/hpack.c [hunk 2]
```diff
@@ -277,9 +277,9 @@ struct aws_hpack_context *aws_hpack_context_new(struct aws_allocator *allocator,
     aws_huffman_decoder_init(&context->decoder, hpack_coder);
 
     /* Initialize dynamic table */
-    const size_t buffer_size = max_dynamic_elements * sizeof(struct aws_http_header);
-    context->dynamic_table.buffer = aws_mem_acquire(allocator, buffer_size);
-    memset(context->dynamic_table.buffer, 0, buffer_size);
+    if (max_dynamic_elements) {
+        context->dynamic_table.buffer = aws_mem_calloc(allocator, max_dynamic_elements, sizeof(struct aws_http_header));
```
Comment (graebm) [inline source/hpack.c]: not checking the result of this allocation

Comment (graebm) [reply source/hpack.c]: https://github.com/awslabs/aws-c-http/pull/121

**Pull request merged**